### ISSUE TITLE
How to use Premium Plan for users created with New-NavContainerNavUser?

### ISSUE THREAD

**Issue by RafaelKoch** [microsoft/navcontainerhelper#716]
Hi,

on a container created with the flag -AssignPremiumPlan still all users created using New-NavContainerNavUser are only assigned Essential access.
I have found no way around, at least not an obvious one.

What am I missing? Hard to believe it is a bug, somebody would have noticed in the long time navcontainerhelper exists, right?

**Comment by NAVspecialist** [comment]
How does your New-NAVcontainer look like? 

(it is clearly a kind of a bug on your side :) )

**Comment by RafaelKoch** [comment]
If You are hinting on forgetting `-AssignPremiumPlan` : not forgotten. :-)

Basically like this, most Parameters are set as required in the build process. But I guess for that matter they are not really important:

```powershell
        New-NavContainer -accept_eula `
                        -accept_outdated `
                        -containerName $ContainerName `
                        -imageName $ImageName `
                        -licenseFile $LicenseFile `
                        -isolation $Isolation `
                        -Credential $credentials `
                        -doNotExportObjectsToText `
                        -enableSymbolLoading:$EnableSymbolLoading `
                        -alwaysPull `
                        -shortcuts "Desktop" `
                        -auth $Auth `
                        -additionalParameters $additionalParameters `
                        -memoryLimit $RAM `
                        -assignPremiumPlan `
                        -updateHosts `
                        -useBestContainerOS `
                        -myScripts $myscripts
```

**Comment by BozidarP** [comment]
You can shoted in table USer Plan directly PremiumPlan :)

**Comment by freddydk** [comment]
> You can shoted in table USer Plan directly PremiumPlan :) (for each user)
> You need SSMS to connect to DB. (IP from contaienr)
> select table [User] (see User Security ID)
> Security ID=$UserID
> With this script:
> INSERT INTO [dbo].[User Plan$63ca2fa4-4f03-4f2b-a480-172fef340d3f] ([Plan ID],[User Security ID]) VALUES ('{8e9002c0-a1d8-4465-b952-817d2948e6e2}','$UserID')

That is what -assignPremiumPlan is doing, so you can check to see whether it has happened?

**Comment by freddydk** [comment]
The code in New-NavContainer for assignpremiumplan is:
    if ($assignPremiumPlan -and !$restoreBakFolder) {
        if (!(Test-Path -Path "$myfolder\SetupNavUsers.ps1")) {
            ('# Invoke default behavior
              . (Join-Path $runPath $MyInvocation.MyCommand.Name)
            ') | Set-Content -Path "$myfolder\SetupNavUsers.ps1"
        }
     
        if ($version.Major -ge 15) {
            $userPlanTableName = 'User Plan$63ca2fa4-4f03-4f2b-a480-172fef340d3f'
        }
        else {
            $userPlanTableName = 'User Plan'
        }
        ('
Get-NavServerUser -serverInstance $ServerInstance -tenant default |? LicenseType -eq "FullUser" | ForEach-Object {
    $UserId = $_.UserSecurityId
    Write-Host "Assign Premium plan for $($_.Username)"
    $dbName = $DatabaseName
    if ($multitenant) {
        $dbName = $TenantId
    }
    $userPlanTableName = '''+$userPlanTableName+'''
    Invoke-Sqlcmd -ErrorAction Ignore -ServerInstance ''localhost\SQLEXPRESS'' -Query "USE [$DbName]
    INSERT INTO [dbo].[$userPlanTableName] ([Plan ID],[User Security ID]) VALUES (''{8e9002c0-a1d8-4465-b952-817d2948e6e2}'',''$userId'')"
}
') | Add-Content -Path "$myfolder\SetupNavUsers.ps1"

So, if you override SetupNavUsers (after container creation) or you specify a .bakfile - then this code doesn't run.
Have a look in SetupNavUsers in your my folder.

**Comment by MJECloud** [comment]
The user which is created with ``` New-NAVContainer ``` has got the premium plan assigned. 
When you create a new NAVUser with ``` New-NavContainerNavUser ```  the new user has no premium plan assigned and can't switch to it. 

Looking in [New-NavContainerNavUser](https://github.com/microsoft/navcontainerhelper/blob/master/UserHandling/New-NavContainerNavUser.ps1) you can see that the SQL-Statement is missing.

**Comment by freddydk** [comment]
It isn't missing... - it was never there:-)
But we could of course add an -assignpremiumplan to the new-navcontainernavuser as well:-)

**Comment by MJECloud** [comment]
My bad, sorry..
Yeah would be a great idea. :)

**Comment by freddydk** [comment]
Shipped in 0.6.4.18

**Comment by RafaelKoch** [comment]
There seems to be some problem still

```powershell
Assign Premium plan for SACHBEARBEITER1
Exception calling "SecureStringToBSTR" with "1" argument(s): "Value cannot be null.
Parameter name: s"
at <ScriptBlock>, <No file>: line 50
at <ScriptBlock>, <No file>: line 49
##[error]Exception calling "SecureStringToBSTR" with "1" argument(s): "Value cannot be null.
Parameter name: s"
At C:\Program 

Files\WindowsPowerShell\Modules\navcontainerhelper\0.6.4.18\ContainerHandling\Invoke-ScriptInNavContainer.ps1:37 

char:13

+             Invoke-Command -Session $session -ScriptBlock $scriptbloc ...

+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

    + CategoryInfo         : NotSpecified: (:) [ForEach-Object], MethodInvocationException

    + FullyQualifiedErrorId : ArgumentNullException,Microsoft.PowerShell.Commands.ForEachObjectCommand

    + PSComputerName        : 3033e6eb24349a2375bd3cda902ba8cc45222485fb5a988764f6225b873756c2
##[error]PowerShell exited with code '1'.
```

**Comment by MJECloud** [comment]
I think the problem there is the missing databasecredential.

**Comment by freddydk** [comment]
Correct - if you set AssignPremiumPlan, you need to specify databaseCredential

 .Parameter AssignPremiumPlan
  For sandbox containers only, assign Premium plan to user if this switch is included
 .Parameter databaseCredential
  Database Credential if using AssignPremiumPlan with foreign database connection

I will add a better error message if it isn't specified.

**Comment by freddydk** [comment]
This should be fixed in 0.6.4.19
